### PR TITLE
add stubs for Manager, Logger.root, Logger.manager

### PR DESCRIPTION
**Pull request by dod-wdwilcox**
Fixes issue 5159
https://github.com/python/typeshed/issues/5159

Please see https://github.com/python/cpython/blob/master/Lib/logging/__init__.py#L1925 for comparison for how this is implemented in standard library.

### CODE REVIEW

File: stdlib/logging/__init__.pyi [hunk 1]
```diff
@@ -31,13 +31,22 @@ class Filterer(object):
...
+
 class Logger(Filterer):
     name: str
     level: int
     parent: Union[Logger, PlaceHolder]
     propagate: bool
     handlers: list[Handler]
     disabled: int
+    root: RootLogger  # undocumented
+    manager: Manager  # undocumented
```
Comment (Akuli) [inline stdlib/logging/__init__.pyi:49]: These are set on the class, not on each instance:

```suggestion
    root: ClassVar[RootLogger]  # undocumented
    manager: ClassVar[Manager]  # undocumented
```

**Review by Akuli — APPROVED** [review]
`Manager` is missing some attributes, but I don't think it matters much since they are undocumented.

File: stdlib/logging/__init__.pyi [hunk 2]
```diff
@@ -31,13 +31,22 @@ class Filterer(object):
     def removeFilter(self, filter: _FilterType) -> None: ...
     def filter(self, record: LogRecord) -> bool: ...
 
+class Manager(object):
+    root: RootLogger
+    disable: int
+    loggerDict: dict[str, Logger]
+    def __init__(self, rootnode: RootLogger) -> None: ...
+    def getLogger(self, name: str) -> None: ...
```
Comment (Akuli) [inline stdlib/logging/__init__.pyi:39]: Oops, I had forgotten the correct way to handle missing attributes, documented in CONTRIBUTING.md:

```suggestion
    def getLogger(self, name: str) -> None: ...
    def __getattr__(self, name: str) -> Any: ...  # incomplete
```

**Review by Akuli — CHANGES_REQUESTED** [review]

File: stdlib/logging/__init__.pyi (hunk 2 above)
Comment (dod-wdwilcox) [reply stdlib/logging/__init__.pyi:39]: I can go ahead and fill out the rest of the public methods if needed.

**Review by Akuli — APPROVED** [review]

File: stdlib/logging/__init__.pyi [hunk 3]
```diff
@@ -31,13 +31,27 @@ class Filterer(object):
     def removeFilter(self, filter: _FilterType) -> None: ...
     def filter(self, record: LogRecord) -> bool: ...
 
+class Manager(object):
+    root: RootLogger
+    disable: int
+    emittedNoHandlerWarning: bool
+    loggerDict: dict[str, Logger]
+    loggerClass: Optional[Logger]
+    logRecordFactory: Optional[LogRecord]
```
Comment (Akuli) [inline stdlib/logging/__init__.pyi:40]: `loggerClass` should be a class inheriting from `Logger` not literally a `Logger` instance. Also, `logRecordFactory` should be typed similarly to global `getLogRecordFactory` and `setLogRecordFactory` functions:

```suggestion
    loggerClass: Optional[Type[Logger]]
    logRecordFactory: Callable[..., LogRecord]
```

This also applies to `setLoggerClass` and `setLogRecordFactory`.

**Review by Akuli — CHANGES_REQUESTED** [review]
Somehow I managed to again miss something when reviewing, lol. Sorry about this.

File: stdlib/logging/__init__.pyi [hunk 4]
```diff
@@ -31,13 +31,27 @@ class Filterer(object):
...
 
+class Manager(object):
+    root: RootLogger
+    disable: int
+    emittedNoHandlerWarning: bool
+    loggerDict: dict[str, Type[Logger]]
+    loggerClass: Optional[Type[Logger]]
+    logRecordFactory: Optional[Callable[..., LogRecord]]
+    def __init__(self, rootnode: RootLogger) -> None: ...
+    def getLogger(self, name: str) -> Type[Logger]: ...
```
Comment (Akuli) [inline stdlib/logging/__init__.pyi:42]: This returns a `Logger` instance, not a class. `PlaceHolder`s are never returned.

```suggestion
    def getLogger(self, name: str) -> Logger: ...
```

File: stdlib/logging/__init__.pyi [hunk 5]
```diff
@@ -31,13 +31,27 @@ class Filterer(object):
     def removeFilter(self, filter: _FilterType) -> None: ...
     def filter(self, record: LogRecord) -> bool: ...
 
+class Manager(object):
+    root: RootLogger
+    disable: int
+    emittedNoHandlerWarning: bool
+    loggerDict: dict[str, Type[Logger]]
```
Comment (Akuli) [inline stdlib/logging/__init__.pyi:38]: Values can't be classes (`Type[Logger]`), but they can be `Logger` instances or `PlaceHolder` instances:

```suggestion
    loggerDict: dict[str, Union[Logger, PlaceHolder]]
```

File: tests/mypy_test.py [hunk 6]
```diff
@@ -16,6 +16,7 @@
 import os
 import re
 import sys
+
```
Comment (Akuli) [inline tests/mypy_test.py:19]: Perhaps we shouldn't isort this file in a logging-related pull request?

**Review by Akuli — CHANGES_REQUESTED** [review]

**Review by Akuli — APPROVED** [review]

**Pull request merged**